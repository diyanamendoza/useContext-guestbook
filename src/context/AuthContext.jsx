import { createContext, useState, useContext } from "react";

const AuthContext = createContext()

const AuthProvider = ({ children }) => {
    const [user, setUser] = useState('')
    const login = (username, password) => {
        const loginOk = username === process.env.AUTH_USER && password === process.env.AUTH_PASS
        if (loginOk) setUser({ username })
        return loginOk
    }
    const logout = () => {
        setUser(null)
    }

    return (
        <AuthContext.Provider value={{ user, login, logout }}> {children} </AuthContext.Provider>
    )
}

const useAuth = () => {
    const context = useContext(AuthContext)
    if (context === undefined) {
        throw new Error('useAuth not used inside provider')
    }
    return context
}

export { AuthProvider, useAuth }