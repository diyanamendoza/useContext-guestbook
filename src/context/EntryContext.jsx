import { useContext, createContext, useState } from 'react'

const EntryContext = createContext()

const EntryProvider = ({ children }) => {
    const [entry, setEntries] = useState([])
    return <EntryContext.Provider value={{ entry, setEntries }}>{children}</EntryContext.Provider>
}

const useEntries = () => {
    const context = useContext(EntryContext)
    if (context === undefined) {
        throw new Error('The useEntries hook is not being called inside a EntryContext.Provider')
    }
    return context
}

export { EntryProvider, useEntries }