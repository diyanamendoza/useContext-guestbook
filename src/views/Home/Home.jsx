import EntryList from "../../components/EntryList/EntryList"
import GuestBook from "../../components/GuestBook/GuestBook"

export default function Home() {
    return (
        <div>
           <GuestBook /> 
           <EntryList />
        </div>
    )
}
