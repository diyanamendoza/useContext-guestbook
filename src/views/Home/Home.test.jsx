import { render } from '@testing-library/react'
import { EntryProvider } from '../../context/EntryContext'
import { UserProvider } from '../../context/UserContext'
import Home from './Home'

it('should render Home', () => {
  const { container } = render(
    <EntryProvider>
    <UserProvider>
        <Home />
     </UserProvider>
    </EntryProvider>)
  expect(container).toMatchSnapshot()
})