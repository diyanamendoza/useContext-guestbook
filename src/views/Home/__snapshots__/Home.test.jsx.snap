// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home 1`] = `
<div>
  <div>
    <h2>
      Please sign the guestbook!
    </h2>
    <form>
      <div
        class="nameinput"
      >
        <label
          for="username"
        >
          Your Name
        </label>
        <input
          id="username"
          placeholder="name"
          type="text"
          value=""
        />
      </div>
      <label
        for="guest-entry"
      >
        Guest Entry
      </label>
      <textarea
        id="guest-entry"
        placeholder="Your message"
      />
      <button
        type="submit"
      >
        Sign
      </button>
      
    </form>
    <ul />
  </div>
</div>
`;
